src/slf4ec.c

```diff
@@ -166,68 +168,26 @@ LogResult noLog()
-            va_list vaList;                                                             \
-            va_start(vaList, formatStr);                                                \
-                                                                                        \
-            _privateLog(mFile, mLine, mFunction, category, &level, formatStr, &vaList); \
-                                                                                        \
-            va_end(vaList);                                                             \
-        }                                                                               \
-    }                                                                                   \
-    else                                                                                \
-    {                                                                                   \
-        returnCode = LOG_NOT_INITIALIZED;                                               \
-    }                                                                                   \
-                                                                                        \
-    return returnCode
-
-/**
- * Function to avoid duplicate code inside nfLog1() and yfLog1().
- */
-inline static LogResult handleLog0(const char* const file,
-                                   const uint32_t* const line,
-                                   const char* const function,
-                                   const LogCategory* const category,
-                                   const uint8_t* const level,
-                                   const char* const formatStr,
-                                   va_list* const vaList)
-{
-    LogResult returnCode = LOG_OK;
-    if (isInitialized)
-    {
-        if (isCategoryActive(category, level))
-        {
-            _privateLog(file, line, function, category, level, formatStr, vaList);
-        }
-    }
-    else
-    {
-        returnCode = LOG_NOT_INITIALIZED;
-    }
-
-    return returnCode;
-}
-
 #ifndef USE_LOCATION_INFO
 LogResult nfLog0(const LogCategory* const category, const uint8_t level, const char* const msg)
 {
     va_list vaList;
-    return handleLog0(NULL, NULL, NULL, category, &level, msg, &vaList);
+    return _privateLog(NULL, NULL, NULL, category, &level, msg, vaList);
 }
 
 LogResult nfLog1(const LogCategory* const category, const uint8_t level, const char* const formatStr, ...)
 {
-    // SONAR has some weird troubles handling the HANDLE_LOG macro.
-    HANDLE_LOG1(NULL, NULL, NULL);
+    LogResult returnCode;
+    va_list vaList;
+    va_start(vaList, formatStr);
+    returnCode = _privateLog(NULL, NULL, NULL, category, &level, formatStr, vaList);
+    va_end(vaList);
+    return returnCode;
+}
+
+LogResult nfLogv(const LogCategory* category, const uint8_t level, const char* formatStr, va_list vaList)
+{
+    return _privateLog(NULL, NULL, NULL, category, &level, formatStr, vaList);
```

Comment:
My understanding is that you should call va_copy() and va_end() between each function calls, as the whole reason why va_copy() exists is to support implementations where va_list contains an offset to the stack pointer, which changes between each function call.

Might be simpler to pass a pointer to va_list instead.